apps/labrinth/src/queue/payouts/mural.rs

```diff
@@ -0,0 +1,183 @@
+                        // only Polygon chain is currently supported
+                        blockchain: muralpay::Blockchain::Polygon,
+                        wallet_address,
+                    },
+                }
+            }
+        };
+
+        let payout = muralpay::CreatePayout {
+            amount,
+            payout_details,
+            recipient_info,
+            supporting_details: None,
+            // TODO
+            // Some(muralpay::SupportingDetails {
+            //     supporting_document: Some(todo!()),
+            //     payout_purpose: Some(muralpay::PayoutPurpose::VendorPayment),
+            // }),
+        };
+
+        let payout_request = muralpay
+            .client
+            .create_payout_request(
+                muralpay.source_account_id,
+                Some(format!("User {user_id}")),
+                &[payout],
+            )
+            .await
+            .map_err(|err| match err {
+                MuralError::Api(err) => ApiError::Request(err.into()),
+                err => ApiError::Internal(err.into()),
+            })?;
+
+        // try to immediately execute the payout request...
+        let result = async move {
+            muralpay
+                .client
+                .execute_payout_request(payout_request.id)
+                .await
+                .wrap_internal_err("failed to execute payout request")?;
+            eyre::Ok(())
+        }
+        .await;
+
+        // and if it fails, make sure to immediately cancel it -
+        // we don't want floating payout requests
+        if let Err(err) = result {
+            muralpay
+                .client
+                .cancel_payout_request(payout_request.id)
+                .await
+                .wrap_internal_err(
+                    "failed to cancel unexecuted payout request",
+                )?;
+            return Err(ApiError::Internal(err));
+        }
+
+        Ok(payout_request)
+    }
+
+    pub async fn cancel_muralpay_payout_request(
+        &self,
+        id: muralpay::PayoutRequestId,
+    ) -> Result<()> {
+        let muralpay = self.muralpay.load();
+        let muralpay = muralpay
+            .as_ref()
+            .wrap_err("Mural Pay client not available")?;
+
+        muralpay.client.cancel_payout_request(id).await?;
+        Ok(())
+    }
+
+    pub async fn get_mural_balance(&self) -> Result<Option<AccountBalance>> {
+        let muralpay = self.muralpay.load();
+        let muralpay = muralpay
+            .as_ref()
+            .wrap_err("Mural Pay client not available")?;
+
+        let account = muralpay
+            .client
+            .get_account(muralpay.source_account_id)
+            .await?;
+        let details = account
+            .account_details
+            .wrap_err("source account does not have details")?;
+        let available = details
+            .balances
+            .iter()
+            .map(|balance| {
+                if balance.token_symbol == "USDC" {
```

Comment:
It seems we hardcode `"USDC"` as a token symbol through this file several times, and from a quick reading of the code it's not immediately clear what's that special about USDC. What if we decide to maintain a balance in USDT? Or switch to USDT-only balances it later on? Extracting `"USDC"` to a constant would help explaining the role `USDC` has and streamline potential stablecoin migrations we may do in the future.